js/src/creatives/mute.js

```diff
@@ -0,0 +1,77 @@
+import FW from '../fw/fw';
+import VASTPLAYER from '../players/vast-player';
+import CONTENTPLAYER from '../players/content-player';
+import VPAID from '../players/vpaid';
+
+const MUTE = {};
+
+const _setMute = function (muted) {
+
+  if (typeof muted !== 'boolean') {
+    return;
+  }
+  if (this.adOnStage && this.adIsLinear) {
+    if (this.isVPAID) {
+      if (muted) {
+        VPAID.setAdVolume.call(this, 0);
+      } else {
+        VPAID.setAdVolume.call(this, 1);
+      }
+    } else {
+      VASTPLAYER.setMute.call(this, muted);
+    }
+  }
+
+  CONTENTPLAYER.setMute.call(this, muted);
+};
+
+const _getMute = function () {
+  if (this.adOnStage && this.adIsLinear) {
+    if (this.isVPAID) {
+      if (VPAID.getAdVolume.call(this) === 0) {
+        return true;
+      }
+      return false;
+    } else {
+      return VASTPLAYER.getMute.call(this);
+    }
+  }
+  
+  return CONTENTPLAYER.getMute.call(this);
+};
+
+const _onClickMute = function (event) {
+  if (event) {
```

Comment:
do we need this conditional check? `addEventListener` will always pass an event object.